server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java

```diff
@@ -260,6 +260,7 @@ private void handleEvents() throws InterruptedException {
                         if (deadlineMap.isEmpty() && (shuttingDown || interrupted)) {
                             // If there are no more entries to process, and the queue is
                             // closing, exit the thread.
+                            lock.unlock();
```

Comment:
it seems the lock can get freed by the `finally` block, right?

https://github.com/apache/kafka/blob/trunk/server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java#L293

Reply:
You are right. Close this.